pkg/tsdb/influxdb/flux/executor.go

```diff
@@ -96,6 +97,10 @@ func readDataFrames(logger log.Logger, result *api.QueryTableResult, maxPoints i
 		err := builder.Append(result.Record())
 		if err != nil {
 			dr.Error = err
+			var maxSeriesError maxSeriesExceededError
+			if errors.As(err, &maxSeriesError) {
```

Comment:
oh wow! i didnt know about `errors.As` 
if I understand correctly, is this ensuring the error is a `maxSeriesError`?

Reply:
[yep](https://pkg.go.dev/errors#As)!